In [ ]:
# Biblioteca científica
import numpy as np
# Biblioteca para análise de dados
import pandas as pd
# Suprimindo os warnings da biblioteca pandas
pd.options.mode.chained_assignment = None
# Bibliotecas de visualização de dados
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches
import matplotlib.ticker as mtick
# import missingno as msno
plt.style.use('ggplot')
import seaborn as sns
sns.set_style('whitegrid')
%matplotlib inline

In [9]:
# Custo das universidades federais do nordeste
custo_ufs = pd.read_csv('../data/custo_ies_federais_nordeste.csv')
# Exibindo o dataframe
custo_ufs

Instituicao   Sigla  \
0           UNIVERSIDADE FEDERAL DE PERNAMBUCO (UFPE)    UFPE   
1              UNIVERSIDADE FEDERAL DE ALAGOAS (UFAL)    UFAL   
2                UNIVERSIDADE FEDERAL DA BAHIA (UFBA)    UFBA   
3       UNIVERSIDADE FEDERAL DE CAMPINA GRANDE (UFCG)    UFCG   
4    UNIVERSIDADE FEDERAL RURAL DE PERNAMBUCO (UFRPE)   UFRPE   
5                 UNIVERSIDADE FEDERAL DO CEARÁ (UFC)     UFC   
6                UNIVERSIDADE FEDERAL DO PIAUÍ (UFPI)    UFPI   
7              UNIVERSIDADE FEDERAL DA PARAÍBA (UFPB)    UFPB   
8             UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)    UFMA   
9   UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE (U...    UFRN   
10              UNIVERSIDADE FEDERAL DE SERGIPE (UFS)     UFS   
11  UNIVERSIDADE FEDERAL RURAL DO SEMI-ÁRIDO (UFERSA)  UFERSA   

    despesaLiquidada  totalAlunos  custoAluno  
0             1557.9        38774     39754.9  
1              802.8        23741     33813.6  
2             1480.0        41445     35709.2  
3              661.4        17194     38468.7  
4              579.0        13989     41387.5  
5             1560.1        32269     48346.1  
6              776.2        30348     25575.5  
7             1551.5        34084     45518.8  
8              729.5        32204     22652.6  
9             1694.5        32872     51548.1  
10             635.4        30060     21138.2  
11             253.9         9757     26027.1

In [123]:
# Custo de cada aluno por universidade federal no nordeste
custo_aluno = custo_ufs[['Instituicao', 'custoAluno']]

# Adicionando a coluna de média no enade por curso com valor default zero
custo_aluno['NT_GER'] = 0

# Exibindo o dataframe
custo_aluno

Instituicao  custoAluno  NT_GER
0           UNIVERSIDADE FEDERAL DE PERNAMBUCO (UFPE)     39754.9       0
1              UNIVERSIDADE FEDERAL DE ALAGOAS (UFAL)     33813.6       0
2                UNIVERSIDADE FEDERAL DA BAHIA (UFBA)     35709.2       0
3       UNIVERSIDADE FEDERAL DE CAMPINA GRANDE (UFCG)     38468.7       0
4    UNIVERSIDADE FEDERAL RURAL DE PERNAMBUCO (UFRPE)     41387.5       0
5                 UNIVERSIDADE FEDERAL DO CEARÁ (UFC)     48346.1       0
6                UNIVERSIDADE FEDERAL DO PIAUÍ (UFPI)     25575.5       0
7              UNIVERSIDADE FEDERAL DA PARAÍBA (UFPB)     45518.8       0
8             UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)     22652.6       0
9   UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE (U...     51548.1       0
10              UNIVERSIDADE FEDERAL DE SERGIPE (UFS)     21138.2       0
11  UNIVERSIDADE FEDERAL RURAL DO SEMI-ÁRIDO (UFERSA)     26027.1       0

In [128]:
# Nota ENADE de cursos de computação nas Universidades Federais do Nordeste
notas_enade_ufs = pd.read_csv('../data/enade2017_ccc_ne_federais_prep.csv')
# Exibindo o dataframe
notas_enade_ufs.head()

CO_IES                              Instituicao  CO_CATEGAD  CO_CURSO  \
0     548  UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)           1     11452   
1     548  UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)           1     11452   
2     548  UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)           1     11452   
3     548  UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)           1     11452   
4     548  UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)           1     11452   

                NO_CURSO Municipio  UF TP_SEXO  NU_IDADE QE_I01  ...   QE_I18  \
0  CIÊNCIA DA COMPUTAÇÃO  São Luís  MA       M        27      A  ...        A   
1  CIÊNCIA DA COMPUTAÇÃO  São Luís  MA       F        22      A  ...        B   
2  CIÊNCIA DA COMPUTAÇÃO  São Luís  MA       M        27      A  ...        A   
3  CIÊNCIA DA COMPUTAÇÃO  São Luís  MA       M        32      A  ...        A   
4  CIÊNCIA DA COMPUTAÇÃO  São Luís  MA       M        25      A  ...        A   

  QE_I19 QE_I20 QE_I21 QE_I22 QE_I23 QE_I24 QE_I25 QE_I26 NT_GER  
0      A      C      A      A      D      A      E      A   45.2  
1      F      K      A      B      B      A      H      F   48.4  
2      A      B      A      A      A      E      H      I   55.5  
3      B      A      A      A      A      A      E      A   23.5  
4      B      G      A      C      E      E      E      F   54.3  

[5 rows x 34 columns]

In [124]:
# Todas as universidades do nordeste
universidades = custo_aluno['Instituicao']

#Exibindo as universidades
universidades

0             UNIVERSIDADE FEDERAL DE PERNAMBUCO (UFPE)
1                UNIVERSIDADE FEDERAL DE ALAGOAS (UFAL)
2                  UNIVERSIDADE FEDERAL DA BAHIA (UFBA)
3         UNIVERSIDADE FEDERAL DE CAMPINA GRANDE (UFCG)
4      UNIVERSIDADE FEDERAL RURAL DE PERNAMBUCO (UFRPE)
5                   UNIVERSIDADE FEDERAL DO CEARÁ (UFC)
6                  UNIVERSIDADE FEDERAL DO PIAUÍ (UFPI)
7                UNIVERSIDADE FEDERAL DA PARAÍBA (UFPB)
8               UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)
9     UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE (U...
10                UNIVERSIDADE FEDERAL DE SERGIPE (UFS)
11    UNIVERSIDADE FEDERAL RURAL DO SEMI-ÁRIDO (UFERSA)
Name: Instituicao, dtype: object

In [126]:
# Calculando a média das notas do enade por universidade e atualizando o DataFrame
for uf_nome in universidades:
        # DataFrame de cada universidade com as notas de todos os alunos
        notas_universidade = notas_enade_ufs[notas_enade_ufs['Instituicao'] == uf_nome]
        
        # Atualizando a coluna de média do enade de cada universidade
        custo_aluno.loc[(custo_aluno['Instituicao']) == uf_nome, 'NT_GER'] = notas_universidade[['Instituicao', 'NT_GER']].mean()['NT_GER']
        
# Exibindo dataframe com as médias do enade de cada universidade
custo_aluno

Instituicao  custoAluno     NT_GER
0           UNIVERSIDADE FEDERAL DE PERNAMBUCO (UFPE)     39754.9  57.030088
1              UNIVERSIDADE FEDERAL DE ALAGOAS (UFAL)     33813.6  45.818571
2                UNIVERSIDADE FEDERAL DA BAHIA (UFBA)     35709.2  47.088060
3       UNIVERSIDADE FEDERAL DE CAMPINA GRANDE (UFCG)     38468.7  54.243284
4    UNIVERSIDADE FEDERAL RURAL DE PERNAMBUCO (UFRPE)     41387.5  48.657576
5                 UNIVERSIDADE FEDERAL DO CEARÁ (UFC)     48346.1  56.564815
6                UNIVERSIDADE FEDERAL DO PIAUÍ (UFPI)     25575.5  42.929412
7              UNIVERSIDADE FEDERAL DA PARAÍBA (UFPB)     45518.8  49.297872
8             UNIVERSIDADE FEDERAL DO MARANHÃO (UFMA)     22652.6  46.181081
9   UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE (U...     51548.1  57.496667
10              UNIVERSIDADE FEDERAL DE SERGIPE (UFS)     21138.2  47.238462
11  UNIVERSIDADE FEDERAL RURAL DO SEMI-ÁRIDO (UFERSA)     26027.1  49.100000

In [127]:
# Correlação entre custo do aluno e a média do enade das universidades federais do nordeste
custo_aluno.corr()

custoAluno    NT_GER
custoAluno    1.000000  0.732179
NT_GER        0.732179  1.000000

## Conclusão

Com um correlação de 0.732179 entre as duas variáveis, podemos então afirmar que quanto mais custar um aluno nas universidades federais do nordeste, maior tende a ser a nota do mesmo no enade.